File: sbe/okx/okx_sbe_1_0/generated-php/com_okx_sbe/TradesChannelEvent.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class TradesChannelEvent
{
    public const TEMPLATE_ID = 1005;
    public const SCHEMA_ID = 1;
    public const SCHEMA_VERSION = 0;
    public const BLOCK_LENGTH = 62;

    public int|float|array|null $instIdCode = null;
    public int|float|array|null $tsUs = null;
    public int|float|array|null $outTime = null;
    public int|float|array|null $seqId = null;
    public int|float|array|null $pxMantissa = null;
    public int|float|array|null $szMantissa = null;
    public int|float|array|null $tradeId = null;
    public int|float|array|null $count = null;
    public int|float|array|null $side = null;
    public int|float|array|null $pxExponent = null;
    public int|float|array|null $szExponent = null;
    public int|float|array|null $source = null;

    public function encode(): string
    {
        $buffer = '';

        if ($this->instIdCode !== null) {
            $buffer .= pack('q', $this->instIdCode);
        }
        if ($this->tsUs !== null) {
            $buffer .= pack('q', $this->tsUs);
        }
        if ($this->outTime !== null) {
            $buffer .= pack('q', $this->outTime);
        }
        if ($this->seqId !== null) {
            $buffer .= pack('q', $this->seqId);
        }
        if ($this->pxMantissa !== null) {
            $buffer .= pack('q', $this->pxMantissa);
        }
        if ($this->szMantissa !== null) {
            $buffer .= pack('q', $this->szMantissa);
        }
        if ($this->tradeId !== null) {
            $buffer .= pack('q', $this->tradeId);
        }
        if ($this->count !== null) {
            $buffer .= pack('s', $this->count);
        }
        if ($this->pxExponent !== null) {
            $buffer .= pack('c', $this->pxExponent);
        }
        if ($this->szExponent !== null) {
            $buffer .= pack('c', $this->szExponent);
        }

        return $buffer;
    }

    public function decode(string $data): void
    {
        $offset = 0;

        $this->instIdCode = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->tsUs = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->outTime = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->seqId = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->pxMantissa = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->szMantissa = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->tradeId = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->count = unpack('s', substr($data, $offset, 2))[1];
        $offset += 2;
        $this->pxExponent = unpack('c', substr($data, $offset, 1))[1];
        $offset += 1;
        $this->szExponent = unpack('c', substr($data, $offset, 1))[1];
        $offset += 1;
    }
}

File: sbe/okx/okx_sbe_1_0/generated-python/com_okx_sbe/TradesChannelEvent.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple, Set
from io import BytesIO

class TradesChannelEvent:
    """SBE message: TradesChannelEvent."""

    TEMPLATE_ID = 1005
    SCHEMA_ID = 1
    SCHEMA_VERSION = 0
    BLOCK_LENGTH = 62

    def __init__(self):
        self.inst_id_code = None
        self.ts_us = None
        self.out_time = None
        self.seq_id = None
        self.px_mantissa = None
        self.sz_mantissa = None
        self.trade_id = None
        self.count = None
        self.side = None
        self.px_exponent = None
        self.sz_exponent = None
        self.source = None

    def encode(self) -> bytes:
        """Encode the message to bytes."""
        buffer = BytesIO()

        if self.inst_id_code is not None:
            buffer.write(struct.pack('<q', self.inst_id_code))
        if self.ts_us is not None:
            buffer.write(struct.pack('<q', self.ts_us))
        if self.out_time is not None:
            buffer.write(struct.pack('<q', self.out_time))
        if self.seq_id is not None:
            buffer.write(struct.pack('<q', self.seq_id))
        if self.px_mantissa is not None:
            buffer.write(struct.pack('<q', self.px_mantissa))
        if self.sz_mantissa is not None:
            buffer.write(struct.pack('<q', self.sz_mantissa))
        if self.trade_id is not None:
            buffer.write(struct.pack('<q', self.trade_id))
        if self.count is not None:
            buffer.write(struct.pack('<h', self.count))
        if self.px_exponent is not None:
            buffer.write(struct.pack('<b', self.px_exponent))
        if self.sz_exponent is not None:
            buffer.write(struct.pack('<b', self.sz_exponent))

        return buffer.getvalue()

    def decode(self, data: bytes) -> None:
        """Decode the message from bytes."""
        pos = 0

        self.inst_id_code = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.ts_us = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.out_time = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.seq_id = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.px_mantissa = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.sz_mantissa = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.trade_id = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.count = struct.unpack_from('<h', data, pos)[0]
        pos += 2
        self.px_exponent = struct.unpack_from('<b', data, pos)[0]
        pos += 1
        self.sz_exponent = struct.unpack_from('<b', data, pos)[0]
        pos += 1

        # Skip to end of block for forward compatibility
        pos = 62

